lib/modules/swarm/upload.js

```diff
@@ -1,50 +1,41 @@
 require('colors');
 let async = require('async');
-let shelljs = require('shelljs');
 
 class Swarm {
   constructor(options) {
     this.options = options;
     this.buildDir = options.buildDir || 'dist/';
   }
 
-  deploy() {
+  deploy(deployOptions) {
     return new Promise((resolve, reject) => {
       console.log("deploying to swarm!");
       let self = this;
+      let web3 = (deployOptions || {}).web3;
       async.waterfall([
-        function findBinary(callback) {
-          let swarm_bin = shelljs.which('swarm');
-
-          if (swarm_bin === 'swarm not found' || !swarm_bin) {
-            console.log('=== WARNING: Swarm not in an executable path. Guessing ~/go/bin/swarm for path'.yellow);
-            swarm_bin = "~/go/bin/swarm";
-          }
-
-          callback(null, swarm_bin);
+        function findWeb3(callback){
+          if(!web3){
+            callback('web3 must be passed in to the swarm deploy() method');
+          }else callback();
         },
-        function runCommand(swarm_bin, callback) {
-          let cmd = `"${swarm_bin}" --defaultpath ${self.buildDir} index.html --recursive up ${self.buildDir}`;
-          console.log(("=== adding " + self.buildDir + " to swarm").green);
-          console.trace(cmd);
-          shelljs.exec(cmd, {silent:true}, function(code, stdout, stderr){ // {silent:true}: don't echo cmd output so it can be controlled via logLevel
-            console.log(stdout.green);
-            callback(stderr, {code: code, output: stdout});
-          });
+        function setProvider(callback){
+          web3.bzz.setProvider(`http://${self.options.storageConfig.host}:${self.options.storageConfig.port}`);
```

Comment:
5. initialize based on storageConfig params

Reply:
Should there be a code change in this file? Or was this comment here for clarity of process/flow?

Reply:
this is fine, although I think it could simply be this.storageConfig and that value be assigned on the constructor (so this method is not coupled with this.options)